pkg/pillar/devicenetwork/devicenetwork.go

```diff
@@ -274,18 +276,90 @@ func devPortInstallAPname(ifname, ap string) {
 		}
 	}
 
-	if len(ap) == 0 {
+	if len(wconfig.APN) == 0 {
 		return
 	}
+
 	file, err := os.Create(filepath)
 	if err != nil {
 		log.Errorln(err)
 		return
 	}
 
-	log.Infof("devPortInstallAPname: write file %s for name %s", filepath, ap)
-	file.WriteString(ap)
+	for _, apn := range wconfig.APN {
+		s := fmt.Sprintf("%s\n", apn)
+		file.WriteString(s)
+		break // only handle the first APN for now utill we know how to handle multiple of APNs
+	}
 	file.Close()
+	log.Infof("devPortInstallAPname: write file %s for name %s", filepath, wconfig.APN)
+}
+
+func devPortInstallWifiConfig(ifname string, wconfig types.WirelessConfig) bool {
```

Comment:
are we generating files for all the configs?
Another question is - If you Pass the location as a parameter Or atleast, make them variables in the Context, you can actually write a go unit test for this. That way, you can test this piece of code very easily..

Can you add some unit tests for this?

Reply:
the file is a default file /config/wpa_supplicant.conf, which we may overwrite with our data. Let me add unit test later to my list.

Reply:
Writing into /config/wpa_supplicant.conf might be fine for an initial implementation.
But we shouldn't do this when we are testing a new configuration since it means that if we get a power failure we might end up starting with the untested /config/wpa_supplicant.conf

If the source of truth comes from the DevicePortConfig and we rebuild wpa_supplicant.conf from that on each boot, then we don't need to keep a copy in /config; just need to make the wlan container pick up the new file.

Reply:
I think you're somehow commenting on the older implementation @nordmark  -- all that is now in /run (where it belongs)